sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -591,20 +590,41 @@ case class FileSourceScanExec(
     logInfo(s"Planning scan with bin packing, max size: $maxSplitBytes bytes, " +
       s"open cost is considered as scanning $openCostInBytes bytes.")
 
+    // Filter files with bucket pruning if possible
+    lazy val bucketingEnabled = fsRelation.sparkSession.sessionState.conf.bucketingEnabled
```

Comment:
There is already a `lazy val bucketedScan: Boolean` in L261

Reply:
@cloud-fan - that's `bucketedScan`, but we need `...conf.bucketingEnabled`, here we already in the code path where `bucketedScan` is false - https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala#L413 .

Reply:
ah sorry I missed that. Probably we can just make it  `val`, since reading conf is very cheap, while `lazy val` has overhead.

Reply:
> since reading conf is very cheap

That's what I feel too, but I got feedback earlier here - https://github.com/apache/spark/pull/31413#discussion_r568257272. @maropu - could you help provide more context here? Thanks.

Reply:
Yea, reverting it back looks okay.

Reply:
Removed `lazy val`.